Unable to compile in several versions of Arduino IDE

I get this error when trying to compile the sketch. Are you able to tell what I'm doing wrong?

<Arduino: 1.8.5 (Windows 10), Board: "Generic ESP8266 Module, 80 MHz, ck, 26 MHz, 40MHz, QIO, 512K (no SPIFFS), v2 Prebuilt (MSS=536), Disabled, None, 115200"

In file included from C:\Users\a_wet\AppData\Local\Temp\arduino_modified_sketch_820655\helloworld.ino:18:0:

C:\Users\a_wet\Documents\Arduino\libraries\httpd/httpServer.h: In member function 'void HttpServer::loop()':

C:\Users\a_wet\Documents\Arduino\libraries\httpd/httpServer.h:140:57: error: no matching function for call to 'min(unsigned int, int&)'

   bp += c.readBytes(buff + bp, min(sizeof(buff) - bp, av));

                                                         ^

C:\Users\a_wet\Documents\Arduino\libraries\httpd/httpServer.h:140:57: note: candidates are:

In file included from c:\users\a_wet\appdata\local\arduino15\packages\esp8266\tools\xtensa-lx106-elf-gcc\1.20.0-26-gb404fb9-2\xtensa-lx106-elf\include\c++\4.8.2\algorithm:62:0,

                 from C:\Users\a_wet\AppData\Local\Arduino15\packages\esp8266\hardware\esp8266\2.4.0\cores\esp8266/Arduino.h:240,

                 from sketch\helloworld.ino.cpp:1:

c:\users\a_wet\appdata\local\arduino15\packages\esp8266\tools\xtensa-lx106-elf-gcc\1.20.0-26-gb404fb9-2\xtensa-lx106-elf\include\c++\4.8.2\bits\stl_algo.h:4226:5: note: template<class _Tp, class _Compare> _Tp std::min(std::initializer_list<_Tp>, _Compare)

     min(initializer_list<_Tp> __l, _Compare __comp)

     ^

c:\users\a_wet\appdata\local\arduino15\packages\esp8266\tools\xtensa-lx106-elf-gcc\1.20.0-26-gb404fb9-2\xtensa-lx106-elf\include\c++\4.8.2\bits\stl_algo.h:4226:5: note:   template argument deduction/substitution failed:

In file included from C:\Users\a_wet\AppData\Local\Temp\arduino_modified_sketch_820655\helloworld.ino:18:0:

C:\Users\a_wet\Documents\Arduino\libraries\httpd/httpServer.h:140:57: note:   mismatched types 'std::initializer_list<_Tp>' and 'unsigned int'

   bp += c.readBytes(buff + bp, min(sizeof(buff) - bp, av));

                                                         ^

In file included from c:\users\a_wet\appdata\local\arduino15\packages\esp8266\tools\xtensa-lx106-elf-gcc\1.20.0-26-gb404fb9-2\xtensa-lx106-elf\include\c++\4.8.2\algorithm:62:0,

                 from C:\Users\a_wet\AppData\Local\Arduino15\packages\esp8266\hardware\esp8266\2.4.0\cores\esp8266/Arduino.h:240,

                 from sketch\helloworld.ino.cpp:1:

c:\users\a_wet\appdata\local\arduino15\packages\esp8266\tools\xtensa-lx106-elf-gcc\1.20.0-26-gb404fb9-2\xtensa-lx106-elf\include\c++\4.8.2\bits\stl_algo.h:4221:5: note: template<class _Tp> _Tp std::min(std::initializer_list<_Tp>)

     min(initializer_list<_Tp> __l)

     ^

c:\users\a_wet\appdata\local\arduino15\packages\esp8266\tools\xtensa-lx106-elf-gcc\1.20.0-26-gb404fb9-2\xtensa-lx106-elf\include\c++\4.8.2\bits\stl_algo.h:4221:5: note:   template argument deduction/substitution failed:

In file included from C:\Users\a_wet\AppData\Local\Temp\arduino_modified_sketch_820655\helloworld.ino:18:0:

C:\Users\a_wet\Documents\Arduino\libraries\httpd/httpServer.h:140:57: note:   mismatched types 'std::initializer_list<_Tp>' and 'unsigned int'

   bp += c.readBytes(buff + bp, min(sizeof(buff) - bp, av));

                                                         ^

In file included from c:\users\a_wet\appdata\local\arduino15\packages\esp8266\tools\xtensa-lx106-elf-gcc\1.20.0-26-gb404fb9-2\xtensa-lx106-elf\include\c++\4.8.2\algorithm:61:0,

                 from C:\Users\a_wet\AppData\Local\Arduino15\packages\esp8266\hardware\esp8266\2.4.0\cores\esp8266/Arduino.h:240,

                 from sketch\helloworld.ino.cpp:1:

c:\users\a_wet\appdata\local\arduino15\packages\esp8266\tools\xtensa-lx106-elf-gcc\1.20.0-26-gb404fb9-2\xtensa-lx106-elf\include\c++\4.8.2\bits\stl_algobase.h:239:5: note: template<class _Tp, class _Compare> const _Tp& std::min(const _Tp&, const _Tp&, _Compare)

     min(const _Tp& __a, const _Tp& __b, _Compare __comp)

     ^

c:\users\a_wet\appdata\local\arduino15\packages\esp8266\tools\xtensa-lx106-elf-gcc\1.20.0-26-gb404fb9-2\xtensa-lx106-elf\include\c++\4.8.2\bits\stl_algobase.h:239:5: note:   template argument deduction/substitution failed:

In file included from C:\Users\a_wet\AppData\Local\Temp\arduino_modified_sketch_820655\helloworld.ino:18:0:

C:\Users\a_wet\Documents\Arduino\libraries\httpd/httpServer.h:140:57: note:   deduced conflicting types for parameter 'const _Tp' ('unsigned int' and 'int')

   bp += c.readBytes(buff + bp, min(sizeof(buff) - bp, av));

                                                         ^

In file included from c:\users\a_wet\appdata\local\arduino15\packages\esp8266\tools\xtensa-lx106-elf-gcc\1.20.0-26-gb404fb9-2\xtensa-lx106-elf\include\c++\4.8.2\algorithm:61:0,

                 from C:\Users\a_wet\AppData\Local\Arduino15\packages\esp8266\hardware\esp8266\2.4.0\cores\esp8266/Arduino.h:240,

                 from sketch\helloworld.ino.cpp:1:

c:\users\a_wet\appdata\local\arduino15\packages\esp8266\tools\xtensa-lx106-elf-gcc\1.20.0-26-gb404fb9-2\xtensa-lx106-elf\include\c++\4.8.2\bits\stl_algobase.h:193:5: note: template<class _Tp> const _Tp& std::min(const _Tp&, const _Tp&)

     min(const _Tp& __a, const _Tp& __b)

     ^

c:\users\a_wet\appdata\local\arduino15\packages\esp8266\tools\xtensa-lx106-elf-gcc\1.20.0-26-gb404fb9-2\xtensa-lx106-elf\include\c++\4.8.2\bits\stl_algobase.h:193:5: note:   template argument deduction/substitution failed:

In file included from C:\Users\a_wet\AppData\Local\Temp\arduino_modified_sketch_820655\helloworld.ino:18:0:

C:\Users\a_wet\Documents\Arduino\libraries\httpd/httpServer.h:140:57: note:   deduced conflicting types for parameter 'const _Tp' ('unsigned int' and 'int')

   bp += c.readBytes(buff + bp, min(sizeof(buff) - bp, av));

                                                         ^

exit status 1
Error compiling for board Generic ESP8266 Module.

This report would have more information with
"Show verbose output during compilation"
option enabled in File -> Preferences.
>

--- a/arduino_libs/httpd/httpServer.h
+++ b/arduino_libs/httpd/httpServer.h
@@ -137,7 +137,7 @@ void HttpServer::loop() {
 
   // Read data up to buffer length
   DEBUG_DUMP(bp);
-  bp += c.readBytes(buff + bp, min(sizeof(buff) - bp, av));
+  bp += c.readBytes(buff + bp, _min(sizeof(buff) - bp, av));
   DEBUG(Serial.print("Buffer state ["));
   DEBUG(Serial.write(buff, bp));
   DEBUG(Serial.println("]"));

--- a/c2_prog_wifi.ino
+++ b/c2_prog_wifi.ino
@@ -27,18 +27,16 @@ const char *c2_print_status_by_name(uint8_t ch) {
 //////////////////////
 class HelloWorldPage : public HttpServerPage {
     void onEndOfHeader() {
-      c->write("<!DOCTYPE html>\
-<html>\
-<body>\
-\
-<form action=\"upload\" method=\"post\" enctype=\"multipart/form-data\">\
-    Select image to upload:\
-    <input type=\"file\" name=\"fileToUpload\" id=\"fileToUpload\">\
-    <input type=\"submit\" value=\"Upload Image\" name=\"submit\">\
-</form>\
-\
-</body>\
-</html>");
+      c->write("<!DOCTYPE html>"
+               "<html>"
+               "<body>"
+               "<form action=\"upload\" method=\"post\" enctype=\"multipart/form-data\">"
+               "Select image to upload:"
+               "<input type=\"file\" name=\"fileToUpload\" id=\"fileToUpload\">"
+               "<input type=\"submit\" value=\"Upload Image\" name=\"submit\">"
+               "</form>"
+               "</body>"
+               "</html>");
       c->stop();
     }
 } helloWorldPage;

